### PR TITLE
fix the bug that unique key doesn't work

### PR DESCRIPTION
**Pull request by bbbearxyz**
## What type of PR is this?

- [ ] API-change
- [ ] BUG
- [ ] Improvement
- [ ] Documentation
- [ ] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue #5790

## What this PR does / why we need it:
fix the bug that unique key doesn't work

### CODE REVIEW

**Comment by bbbearxyz** [comment]
@matrix-meow  rerun failed

File: pkg/sql/colexec/index_table.go [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+	if err != nil {
+		return nil, err
+	}
+	r := vector.MustTCols[types.Rowid](readBat.Vecs[0])
+	newVec := vector.New(readBat.Vecs[0].Typ)
+	switch v.Typ.Oid {
+	case types.T_bool:
+		s := vector.MustTCols[bool](v)
+		ns := vector.MustTCols[bool](readBat.Vecs[1])
+		for _, b := range s {
```
Comment (fengttt) [inline pkg/sql/colexec/index_table.go:46]: What is the size of s and ns?   1M * 1M, this loop will take forever.

**Review by fengttt — CHANGES_REQUESTED** [review]

**Review by fengttt — COMMENTED** [review]
BuildRowidBatch, I don't understand the logic.   This function needs a comment on what it does.

Actually, I think the loop is wrong.

File: pkg/sql/colexec/index_table.go (hunk 1 above)
Comment (bbbearxyz) [reply pkg/sql/colexec/index_table.go:46]: the size of n and ns is the delete batch size and the table row size, I will optimize the code later

**Comment by bbbearxyz** [comment]
> BuildRowidBatch, I don't understand the logic. This function needs a comment on what it does.
> 
> Actually, I think the loop is wrong.

the row batch is use the delete batch to find the row id. when the pkey value is equal, I will get the row id and put it in the newBatch.

**Comment by fengttt** [comment]
> > BuildRowidBatch, I don't understand the logic. This function needs a comment on what it does.
> > Actually, I think the loop is wrong.
> 
> the row batch is use the delete batch to find the row id. when the pkey value is equal, I will get the row id and put it in the newBatch.

At least the 2 loops are not acceptible.    Has to be a binary search or sort merge or hash, something reasonably efficient.

**Pull request closed without merging**